contracts/Aavegotchi/facets/ERC1155BuyOrderFacet.sol

```diff
@@ -0,0 +1,264 @@
+        uint256 quantity,
+        uint256 time
+    );
+
+    function getERC1155BuyOrder(uint256 _buyOrderId) external view returns (ERC1155BuyOrder memory buyOrder_) {
+        buyOrder_ = s.erc1155BuyOrders[_buyOrderId];
+        require(buyOrder_.timeCreated != 0, "ERC1155BuyOrder: ERC1155 buyOrder does not exist");
+    }
+
+    function getERC1155BuyOrderIdsByTokenId(
+        address _erc1155TokenAddress,
+        uint256 _erc1155TokenId
+    ) external view returns (uint256[] memory buyOrderIds_) {
+        buyOrderIds_ = s.erc1155TokenToBuyOrderIds[_erc1155TokenAddress][_erc1155TokenId];
+    }
+
+    function getERC1155BuyOrdersByTokenId(
+        address _erc1155TokenAddress,
+        uint256 _erc1155TokenId
+    ) external view returns (ERC1155BuyOrder[] memory buyOrders_) {
+        uint256[] memory buyOrderIds = s.erc1155TokenToBuyOrderIds[_erc1155TokenAddress][_erc1155TokenId];
+        uint256 length = buyOrderIds.length;
+        buyOrders_ = new ERC1155BuyOrder[](length);
+        for (uint256 i; i < length; i++) {
+            buyOrders_[i] = s.erc1155BuyOrders[buyOrderIds[i]];
+        }
+    }
+
+    function placeERC1155BuyOrder(
+        address _erc1155TokenAddress,
+        uint256 _erc1155TokenId,
+        uint256 _priceInWei,
+        uint256 _quantity,
+        uint256 _duration
+    ) external {
+        uint256 cost = _quantity * _priceInWei;
+        require(cost >= 1e15, "ERC1155BuyOrder: cost should be 0.001 GHST or larger");
+
+        address sender = LibMeta.msgSender();
+        uint256 category = LibSharedMarketplace.getERC1155Category(_erc1155TokenAddress, _erc1155TokenId);
+        uint256 ghstBalance = IERC20(s.ghstContract).balanceOf(sender);
+
+        uint256 oldBuyOrderId = s.buyerToERC1155BuyOrderId[_erc1155TokenAddress][_erc1155TokenId][sender];
+
+        //If an order exists from this user, first refund the GHST
+        if (oldBuyOrderId != 0) {
+            ERC1155BuyOrder storage erc1155BuyOrder = s.erc1155BuyOrders[oldBuyOrderId];
+            require(erc1155BuyOrder.timeCreated != 0, "ERC1155BuyOrder: ERC1155 buyOrder does not exist");
+            require((erc1155BuyOrder.cancelled == false) && (erc1155BuyOrder.lastTimePurchased == 0), "ERC1155BuyOrder: Already processed");
+
+            //If the order is still valid, update the order
+            if ((erc1155BuyOrder.duration == 0) || (erc1155BuyOrder.timeCreated + erc1155BuyOrder.duration >= block.timestamp)) {
+                // Transfer GHST
+                uint256 oldCost = erc1155BuyOrder.quantity * erc1155BuyOrder.priceInWei;
+                uint256 difference = cost - oldCost;
+                if (cost > oldCost) {
+                    require(ghstBalance >= difference, "ERC1155BuyOrder: Not enough GHST!");
+
+                    //Transfer the difference to the contract
+                    LibERC20.transferFrom(s.ghstContract, sender, address(this), difference);
+                } else if (cost < oldCost) {
```

Comment:
What will happen if cost == oldCost? @orionstardust Did you explore this case?

Reply:
@cinnabarhorse if cost and oldCost are same, there's no GHST transfer from price and quantity change. Only buy order updated.

Reply:
even if price and quantity changed, cost and oldCost could be same.
For example, old price was 0.1 and quantity was 10 (old cost 1), and now price is 1 and quantity is 1 (cost is 1),
in this case both cost and old cost could be same.